client/components/Cart.js

```diff
@@ -0,0 +1,98 @@
+// LIBRARIES
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { getArtworks } from '../redux/artworks';
+import { changeCartItem, removeCartItem } from '../redux/cart'
+
+export class Cart extends Component {
+  render() {
+    const {artworks, cart} = this.props
+    //if you go directly to the cart page, you don't have the artwork, so this is a workaround but we probably need to figure out the right way to do this later.
+    if(artworks.length === 0) {
+      this.props.getArtworks();
+      return (<div>Loading</div>)
+    }
+
+    //if nothing in cart, say something nice:
+    if(cart.length === 0) {
+      return (
+        <div>
+          <h2>Your cart is empty!</h2>
+        </div>
+      )
+    }
+
+    //build quantity select options based on max quantity of each artwork in the cart
+    const quantitySelect = (art) => {
+      const quantitySelection = []
+      for (let i = 1; i <= art.quantity; i++) {
+        quantitySelection.push(<option key={i} value={i}>{i}</option>)
+      }
+      return quantitySelection
+    }
+
+    const changeQuantity = (qty, cartItem) => {
+      let isLoggedIn = false
+      if(this.props.user.id) isLoggedIn = true
+      this.props.changeCartItem({...cartItem, quantity: +qty}, isLoggedIn)
+    }
+
+    const removeItem = (cartItem) => {
+      let isLoggedIn = false
+      if(this.props.user.id) isLoggedIn = true
+      this.props.removeCartItem(cartItem, isLoggedIn)
+    }
+
```

Comment:
made this comment in the redux/cart file, but you dont need to pass isLoggedIn every time if it isn't true, since by default it's false.

Reply:
I responded in redux/cart, but I'll say a little more here: lines 41 and 42 set isLoggedIn to true or false, which needs to be passed to removeCartItem on 43. I don't think it could be left out.